services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java

```diff
@@ -0,0 +1,105 @@
+package com.foodie.user.controllers;
+
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import com.foodie.enums.ResponseKeys;
+import com.foodie.user.model.Role;
+import com.foodie.user.model.service.RoleService;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@RestController
+@CrossOrigin
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/users/roles")
+public class RoleController {
+
+    private final RoleService roleService;
+
+    @PostMapping(produces = "application/json", consumes = "application/json")
+    public ResponseEntity<Map<String, ?>> createRole(@Valid @RequestBody Role role) {
```

Comment:
You probably shouldn't be using wildcards like this, this will increase your code's running time

Reply:
I had to do that because of the exception that am catching. Am not sure there is a way to inform it to return either one of two types like we do in typescript

Reply:
You can have classes which represent the possible return types from this method and have those classes extends a base class. Then you'll make that base class the wildcard here